core/shark_turbine/kernel/_support/indexing.py

```diff
@@ -122,20 +115,31 @@ class Grid(metaclass=_GridMeta, symbolic_shape=None):
     """Grid with bounding symbolic shape information in the type."""
 
     symbolic_shape: ClassVar[Optional[SymbolicShapeExpr]]
+    # TODO: dims should also allow dynamic dimensions.
+    dims: list[int]
```

Comment:
Not quite seeing why this belongs statically carried on the *type* like this. It also crosses a line that I was trying to hold: the type should not depend on an indexing context.

Fwiw, the symbolic shape can have expressions that are just constants.

Reply:
I think what you may be reaching for is an accessor or property of the type which gets a list of dims for the current indexing context (vs pre evaluating that at type construction time).

Reply:
I don't think this is being carried on the type. GridMeta is the type which has symbolic_shape, while Grid is an instance which knows how to resolve these shapes unless I'm understanding what you mean wrong.

Reply:
My bad - I was looking right past a GH folding bar and missed the second class definition. Was really confused.